kedro-datasets/tests/ibis/test_table_dataset.py

```diff
@@ -348,6 +360,60 @@ def test_connection_config(self, mocker, table_dataset, connection_config, key):
         table_dataset.load()
         assert ("ibis", key) in table_dataset._connections
 
+    @pytest.mark.parametrize(
+        ("connection_config", "credentials_config", "key"),
+        [
+            (
+                {"backend": "duckdb", "database": "file.db", "extensions": ["spatial"]},
+                {"user": "admin", "password": "secret"},  # pragma: allowlist secret
+                (
+                    ("backend", "duckdb"),
+                    ("database", "file.db"),
+                    ("extensions", ("spatial",)),
+                    ("password", "secret"),
+                    ("user", "admin"),
+                ),
+            ),
+            (
+                [],
+                {
+                    "host": "xxx.sql.azuresynapse.net",
+                    "database": "xxx",
+                    "query": {"driver": "ODBC Driver 17 for SQL Server"},
+                    "backend": "mssql",
+                },
+                (
+                    ("backend", "mssql"),
+                    ("database", "xxx"),
+                    ("host", "xxx.sql.azuresynapse.net"),
+                    ("query", (("driver", "ODBC Driver 17 for SQL Server"),)),
+                ),
+            ),
+            (
+                None,
+                None,
+                (
+                    ("backend", "duckdb"),
+                    ("database", ":memory:"),
+                ),
+            ),
+        ],
+        indirect=["connection_config", "credentials_config"],
+    )
+    def test_connection_config_with_credentials(
+        self, mocker, table_dataset, connection_config, credentials_config, key
+    ):
+        # Fix: handle non-dict connection_config/credentials_config
+        if isinstance(connection_config, dict) and "backend" in connection_config:
+            backend = connection_config["backend"]
+        elif isinstance(credentials_config, dict) and "backend" in credentials_config:
+            backend = credentials_config["backend"]
```

Comment:
I think this is incorrect, if both specify backend, since `credentials_config` should override `connection_config`. Here, we wouldn't get to the `elif`.

I think it's technically correct if the `credentials_config` condition is first, and the `connection_config` one second.